### DISCUSSION TITLE
Make a nicer file upload for Umbraco.Forms

### DISCUSSION THREAD

**Discussion by graealex** [umbraco/Umbraco.Forms.Issues#1507] [Ideas]
Currently, the file upload is only the bare minimum. 

- You can't do drag and drop.
- You can't select multiple files from different folders. If you want to upload multiple files, they need to be in the same folder, because you can't add any files towards the already done selection, so they need to be selected in one go.
- Every time you click "Choose Files", all pre-existing files get removed. If you click on accident, and then press "Cancel" in the file dialog, again, all files get removed.
- There is no way to manage the selected files - i.e. remove a single one from the list.
- Even if the files you are uploading are just PNGs and JPEGs, you don't get any preview.

For a paid product, this is really poor performance.

**Comment by OwainWilliams** [comment]
Was just looking to log this very same issue.
Uploading multiple files is problematic, even if the uploader allows it.

**Comment by sniffdk** [comment]
Was looking to log another issue and stumbled across this issue from **2022** and we _just_ had to do our own custom implementation of the upload field on a project to support these (by todays standards) basic features.

**Comment by TvGessel** [comment]
Thank you for the suggestion. We have added this to our internal backlog for further discussion and prioritization to determine if and when we can include this in the roadmap.
